<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Negative control example: HDL-&gt;CAD â€¢ MRCorge</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Negative control example: HDL-&gt;CAD">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MRCorge</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/HDL_CAD.html">Negative control example: HDL-&gt;CAD</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/zhwm/MRCORGE/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Negative control example: HDL-&gt;CAD</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zhwm/MRCORGE/blob/main/vignettes/HDL_CAD.Rmd" class="external-link"><code>vignettes/HDL_CAD.Rmd</code></a></small>
      <div class="d-none name"><code>HDL_CAD.Rmd</code></div>
    </div>

    
    
<p>Increased serum HDL levels have been associated with a decreased risk
of CAD in observational studies. However, randomized controlled trials
have shown no meaningful protective effects of HDL-increasing drugs
against CAD in the general population with normal HDL levels. In this
study, we will use the Mendelian randomization analysis of HDL on CAD as
a negative control example to showcase the utility of MR CORGE.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zhwm/MRCORGE" class="external-link">MRCorge</a></span><span class="op">)</span></span></code></pre></div>
<p>First, we load the example data included the MRCorge package. Both
exposure (HDL) and outcome (CAD) GWAS data were harmonized using the
<code>TwoSampleMR</code> package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HDL_CAD"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">HDL_CAD</span><span class="op">)</span></span>
<span><span class="co">#&gt;          SNP effect_allele.exposure other_allele.exposure effect_allele.outcome</span></span>
<span><span class="co">#&gt; 1 rs10019888                      A                     G                     A</span></span>
<span><span class="co">#&gt; 2 rs10087900                      G                     A                     G</span></span>
<span><span class="co">#&gt; 3   rs102275                      T                     C                     T</span></span>
<span><span class="co">#&gt; 4   rs103294                      T                     C                     T</span></span>
<span><span class="co">#&gt; 5 rs10468017                      T                     C                     T</span></span>
<span><span class="co">#&gt; 6  rs1047891                      C                     A                     C</span></span>
<span><span class="co">#&gt;   other_allele.outcome beta.exposure beta.outcome eaf.exposure eaf.outcome</span></span>
<span><span class="co">#&gt; 1                    G        0.0270  -0.00865496       0.8364      0.8386</span></span>
<span><span class="co">#&gt; 2                    A        0.0231   0.00148392       0.5607      0.5718</span></span>
<span><span class="co">#&gt; 3                    C        0.0391   0.01101173       0.6280      0.6455</span></span>
<span><span class="co">#&gt; 4                    C        0.0523  -0.00184047       0.1860      0.2152</span></span>
<span><span class="co">#&gt; 5                    C        0.1179   0.01205890       0.2757      0.2847</span></span>
<span><span class="co">#&gt; 6                    A        0.0269   0.01981174       0.6979      0.6965</span></span>
<span><span class="co">#&gt;   remove palindromic ambiguous id.outcome chr.outcome pos.outcome  se.outcome</span></span>
<span><span class="co">#&gt; 1  FALSE       FALSE     FALSE     7n4jWF           4    26061368 0.006765569</span></span>
<span><span class="co">#&gt; 2  FALSE       FALSE     FALSE     7n4jWF           8   143221248 0.005060500</span></span>
<span><span class="co">#&gt; 3  FALSE       FALSE     FALSE     7n4jWF          11    61790331 0.005160381</span></span>
<span><span class="co">#&gt; 4  FALSE       FALSE     FALSE     7n4jWF          19    54293995 0.006095529</span></span>
<span><span class="co">#&gt; 5  FALSE       FALSE     FALSE     7n4jWF          15    58386313 0.005427939</span></span>
<span><span class="co">#&gt; 6  FALSE       FALSE     FALSE     7n4jWF           2   210675783 0.005448003</span></span>
<span><span class="co">#&gt;   pval.outcome info.outcome samplesize.outcome outcome mr_keep.outcome</span></span>
<span><span class="co">#&gt; 1    0.2008000           NA            1154330 outcome            TRUE</span></span>
<span><span class="co">#&gt; 2    0.7693000           NA            1152300 outcome            TRUE</span></span>
<span><span class="co">#&gt; 3    0.0328500           NA            1164000 outcome            TRUE</span></span>
<span><span class="co">#&gt; 4    0.7627000           NA            1159810 outcome            TRUE</span></span>
<span><span class="co">#&gt; 5    0.0263100           NA            1165600 outcome            TRUE</span></span>
<span><span class="co">#&gt; 6    0.0002764           NA            1159050 outcome            TRUE</span></span>
<span><span class="co">#&gt;   pval_origin.outcome se.exposure pval.exposure info.exposure</span></span>
<span><span class="co">#&gt; 1            reported      0.0046     4.901e-08            NA</span></span>
<span><span class="co">#&gt; 2            reported      0.0036     2.174e-09            NA</span></span>
<span><span class="co">#&gt; 3            reported      0.0035     6.404e-28            NA</span></span>
<span><span class="co">#&gt; 4            reported      0.0044     3.995e-30            NA</span></span>
<span><span class="co">#&gt; 5            reported      0.0038    1.210e-188            NA</span></span>
<span><span class="co">#&gt; 6            reported      0.0039     8.730e-10            NA</span></span>
<span><span class="co">#&gt;   samplesize.exposure exposure mr_keep.exposure pval_origin.exposure</span></span>
<span><span class="co">#&gt; 1              187077 exposure             TRUE             reported</span></span>
<span><span class="co">#&gt; 2              183672 exposure             TRUE             reported</span></span>
<span><span class="co">#&gt; 3              187085 exposure             TRUE             reported</span></span>
<span><span class="co">#&gt; 4              175917 exposure             TRUE             reported</span></span>
<span><span class="co">#&gt; 5              181223 exposure             TRUE             reported</span></span>
<span><span class="co">#&gt; 6              182043 exposure             TRUE             reported</span></span>
<span><span class="co">#&gt;   id.exposure action mr_keep</span></span>
<span><span class="co">#&gt; 1      0GbyEu      2    TRUE</span></span>
<span><span class="co">#&gt; 2      0GbyEu      2    TRUE</span></span>
<span><span class="co">#&gt; 3      0GbyEu      2    TRUE</span></span>
<span><span class="co">#&gt; 4      0GbyEu      2    TRUE</span></span>
<span><span class="co">#&gt; 5      0GbyEu      2    TRUE</span></span>
<span><span class="co">#&gt; 6      0GbyEu      2    TRUE</span></span></code></pre></div>
<p>Next, we use the <code>mrcorge</code> function to perform MR analyses
based on the core gene hypothesis. Specifically, we use the per-allele
effect size (specified with <code>rank</code>) to categorize instrument
variants into 20 groups (specified with <code>K</code>) and perform
inverse variance weighted MR (specified with <code>mr_ivw</code>) within
each group.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mrcorge.html">mrcorge</a></span><span class="op">(</span><span class="va">HDL_CAD</span>, rank<span class="op">=</span><span class="st">'beta'</span>, K<span class="op">=</span><span class="fl">20</span>, method_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mr_ivw"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; TwoSampleMR version 0.6.8</span></span>
<span><span class="co">#&gt;   [&gt;] New authentication requirements: https://mrcieu.github.io/ieugwasr/articles/guide.html#authentication.</span></span>
<span><span class="co">#&gt;   [&gt;] Major upgrades to our servers completed to improve service and stability.</span></span>
<span><span class="co">#&gt;   [&gt;] We need your help to shape our emerging roadmap!</span></span>
<span><span class="co">#&gt;       Please take 2 minutes to give us feedback -</span></span>
<span><span class="co">#&gt;       https://forms.office.com/e/eSr7EFAfCG</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; MR CORGE estimates: </span></span>
<span><span class="co">#&gt;   id.exposure id.outcome outcome exposure                    method nsnp</span></span>
<span><span class="co">#&gt; 1      0GbyEu     7n4jWF outcome exposure Inverse variance weighted    6</span></span>
<span><span class="co">#&gt;             b        se      pval    Fstat group       category</span></span>
<span><span class="co">#&gt; 1 -0.03943587 0.1004654 0.6946652 750.7501     1 Group-specific</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 2 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 3 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 4 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 5 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 6 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 7 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 8 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 9 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 10 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 11 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 12 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 13 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 14 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 15 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 16 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 17 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 18 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 19 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining group-specific estimates for instrument group 20 / 20 ... </span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 1 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 2 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 3 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 4 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 5 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 6 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 7 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 8 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 9 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 10 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 11 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 12 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 13 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 14 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 15 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 16 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 17 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 18 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 19 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span>
<span><span class="co">#&gt; obtaining cumulative estimates for instrument group 20 / 20 ...</span></span>
<span><span class="co">#&gt; Analysing '0GbyEu' on '7n4jWF'</span></span></code></pre></div>
<p>Based on the core instruments, the MR CORGE estimate shows that serum
HDL level may not be causally associated with CAD. As we add more
peripheral instruments, MR results suggest protective effects of HDL on
CAD risk, likely due to horizontal pleiotropy.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_mrcorge.html">plot_mrcorge</a></span><span class="op">(</span><span class="va">res</span>, scale<span class="op">=</span><span class="st">'exp'</span><span class="op">)</span></span></code></pre></div>
<p><img src="HDL_CAD_files/figure-html/plot1-1.png" width="960"></p>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Wenmin Zhang, Tianyuan Lu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
